src/MOL-SLLOD/compute_pressure_mol.cpp

```diff
@@ -371,3 +363,104 @@ void ComputePressureMol::reset_extra_compute_fix(const char *id_new)
+  if (atom->nmax > nmax) {
+    nmax = atom->nmax;
+    memory->grow(com_peratom, nmax, 3, "pressure/mol:com_peratom");
+  }
+  for (int i = 0; i < atom->nghost; ++i) {
+    tagint m = atom->molecule[i]-1;
+    if (m < 0) {
+      com_peratom[i][0] = atom->x[i][0];
+      com_peratom[i][1] = atom->x[i][1];
+      com_peratom[i][2] = atom->x[i][2];
+    } else {
+      com_peratom[i][0] = molprop->com[m][0];
+      com_peratom[i][1] = molprop->com[m][1];
+      com_peratom[i][2] = molprop->com[m][2];
+      // CoM is stored in unwrapped coordinates. Need to map to same image as atom
+      imageint ix = (2*IMGMAX - (atom->image[i] & IMGMASK)) & IMGMASK;
+      imageint iy = (2*IMGMAX - (atom->image[i] >> IMGBITS & IMGMASK)) & IMGMASK;
+      imageint iz = (2*IMGMAX - (atom->image[i] >> IMG2BITS)) & IMGMASK;
+      domain->unmap(com_peratom[i], ix | (iy << IMGBITS) | (iz << IMG2BITS));
+    }
+  }
+}
+
+/* ----------------------------------------------------------------------
+   tally molecular virials into global accumulator
+   have delx, dely, delz and fpair (which gives fx, fy, fz)
+   get delcomx, delcomy, delcomz (molecule centre-of-mass separation)
+   from atom->property_molecule
+------------------------------------------------------------------------- */
+
+void ComputePressureMol::pair_tally_callback(int i, int j, int nlocal,
+    int newton_pair, double evdwl, double ecoul, double fpair,
+    double delx, double dely, double delz)
+{
+  // Virial does not need to be tallied if we didn't do setup this step
+  if (did_setup != update->ntimestep) return;
+
+  double delcom[3], v[9];
+
+  for (int d = 0; d < 3; d++) {
+    delcom[d] = com_peratom[i][d] - com_peratom[j][d];
+  }
+
+  v[0] = delcom[0]*delx*fpair;
+  v[1] = delcom[1]*dely*fpair;
+  v[2] = delcom[2]*delz*fpair;
+  v[3] = delcom[0]*dely*fpair;
+  v[4] = delcom[0]*delz*fpair;
+  v[5] = delcom[1]*delz*fpair;
+  v[6] = delcom[1]*delx*fpair;
+  v[7] = delcom[2]*delx*fpair;
+  v[8] = delcom[2]*dely*fpair;
+
+  if (newton_pair) {
+    pair_virial[0] += v[0];
+    pair_virial[1] += v[1];
+    pair_virial[2] += v[2];
+    pair_virial[3] += v[3];
+    pair_virial[4] += v[4];
+    pair_virial[5] += v[5];
+    pair_virial[6] += v[6];
+    pair_virial[7] += v[7];
+    pair_virial[8] += v[8];
+  } else {
+    if (i < nlocal) {
+      pair_virial[0] += 0.5*v[0];
+      pair_virial[1] += 0.5*v[1];
+      pair_virial[2] += 0.5*v[2];
+      pair_virial[3] += 0.5*v[3];
+      pair_virial[4] += 0.5*v[4];
+      pair_virial[5] += 0.5*v[5];
+      pair_virial[6] += 0.5*v[6];
+      pair_virial[7] += 0.5*v[7];
+      pair_virial[8] += 0.5*v[8];
+    }
+    if (j < nlocal) {
+      pair_virial[0] += 0.5*v[0];
+      pair_virial[1] += 0.5*v[1];
+      pair_virial[2] += 0.5*v[2];
+      pair_virial[3] += 0.5*v[3];
+      pair_virial[4] += 0.5*v[4];
+      pair_virial[5] += 0.5*v[5];
+      pair_virial[6] += 0.5*v[6];
+      pair_virial[7] += 0.5*v[7];
+      pair_virial[8] += 0.5*v[8];
+    }
+  }
```

Comment:
```suggestion

  double mult = (!!(pair_newton) || i < nlocal && j < nlocal) ? 1.0 : 0.5;
  // add to virial twice if pair_newton or both i and j are local
  // and once if !pair_newton AND only one of i and j are local

  for (int d = 0; d < 3; d++) {
    delcom[d] = mult * (com_peratom[i][d] - com_peratom[j][d]);
  }

  v[0] = delcom[0]*delx*fpair;
  v[1] = delcom[1]*dely*fpair;
  v[2] = delcom[2]*delz*fpair;
  v[3] = delcom[0]*dely*fpair;
  v[4] = delcom[0]*delz*fpair;
  v[5] = delcom[1]*delz*fpair;
  v[6] = delcom[1]*delx*fpair;
  v[7] = delcom[2]*delx*fpair;
  v[8] = delcom[2]*dely*fpair;

  for (int dd = 0; dd < 9; dd++) {
    pair_virial[dd] += v[dd];
  }
```

not sure if putting mult in delcom makes the code too unreadable -- but it does reduce the number of operations needed in a fairly inner loop

Reply:
Added this in latest commit ([e4b8a2b](https://github.com/CTCMS-UQ/lammps/pull/11/commits/e4b8a2b3755163f3bed39cb239baccf1e2cd9bf5)). We should also check whether it's faster to just add directly to `pair_virial` rather than using `v` as an intermediate. Could hinder vectorisation though.